### ISSUE TITLE
Inconsistency in image comparison

### ISSUE THREAD

**Issue by neelasha23** [ploomber/sklearn-evaluation#76]
The results of matplotlib's `@image_comparison` are a bit inconsistent sometimes (behaving differently in local vs CI). Maybe we can aim to build a custom utility for comparing images from plots.

**Comment by edublancas** [comment]
I remember encountering problems in the past. Can you share the diff view of the image comparison? (when a test fails, it'll produce a new image with the difference between the two)

When I had that issue, I realized the problems were in the labels. Ensuring that I had the same `freetype` version in local and CI fixed the issue, see here: https://github.com/ploomber/sklearn-evaluation/blob/117e3866709fcb6d91d9aed18560266ef815fab4/.github/workflows/ci.yml#L30

**Comment by neelasha23** [comment]
Ok I will check this. Yes the problem is mostly the label and tick positions

**Comment by edublancas** [comment]
I encountered this problem as well. I found out that this happens with Python 3.6 and 3.7, so I increased the tolerance for the image difference. Here:

https://github.com/ploomber/sklearn-evaluation/blob/c2c7b96c393c0c430dd3c239470583d865c1ed0e/tests/test_grid_search.py#L9

and here:

https://github.com/ploomber/sklearn-evaluation/blob/c2c7b96c393c0c430dd3c239470583d865c1ed0e/tests/test_plot.py#L22

this is already in the master branch, let me know if this fixes it for you as well!

**Comment by neelasha23** [comment]
This didn't work out for me. 
1. Test are failing on all Python versions including Python 3.10.
2. Increased the `tol` value to 5.0 as well but the [CI](https://github.com/ploomber/sklearn-evaluation/actions/runs/3572278267/jobs/6004988230) still fails on version 3.10.

```
image_comparison = partial(_image_comparison,
                           tol=5.0 if sys.version_info.minor in [6, 7, 8, 9, 10] else 0,
                           remove_text=False,
                           extensions=['png'])
```

The issue is with two of the test cases. The remaining cases are fine.
Is there a way to see the RMS Value that the `image_comparison` is generating, and the failed image in the CI?

**Comment by edublancas** [comment]
The RMS value is printed in the logs, I clicked on the CI link you posted and saw this:

- RMS 36.028
- RMS 47.267

```
 =========================== short test summary info ============================
FAILED tests/test_clustering.py::test_string_classes[png] - matplotlib.testing.exceptions.ImageComparisonFailure: images not close (RMS 36.028):
	result_images/test_clustering/string_classes_silhouette.png
	result_images/test_clustering/string_classes_silhouette-expected.png
	result_images/test_clustering/string_classes_silhouette-failed-diff.png
FAILED tests/test_clustering.py::test_ax_params[png] - matplotlib.testing.exceptions.ImageComparisonFailure: images not close (RMS 47.267):
	result_images/test_clustering/ax_params_silhouette.png
	result_images/test_clustering/ax_params_silhouette-expected.png
	result_images/test_clustering/ax_params_silhouette-failed-diff.png
=========== 2 failed, 410 passed, 179 warnings in 119.46s (0:01:59) ============
nox > Command pytest tests/ src/ examples/ --cov=sklearn_evaluation --doctest-modules failed with exit code 1
nox > Session tests failed.
```

Unfortunately, there is no easy way to download the plots. What I've done before is to re-create the environment locally (same python and matplotlib version) and run the tests. Often this is enough to reproduce the error (same RMS).

**Comment by neelasha23** [comment]
While testing this on the metrics PR I realised that this issue is consistently occurring across all PRs which have a testcase that tests the corresponding plot by passing string class name.
I tested by installing freetype 2.12 in my local and created python env 3.8 and 3.10, matplotlib 3.6.2 (same as CI). Testcases are passing in my local in both environments, but fails in all python version in the [CI](https://github.com/ploomber/sklearn-evaluation/actions/runs/3591100132)
RMS Values:
3.8 : 20.890
3.7: 14.15
3.9: 10.034
3.10: 10.034

So as of now only for this function I have added the `tol` parameter. This solves the issue.
